lib/workers/branch/pnpm.js

```diff
@@ -0,0 +1,84 @@
+const fs = require('fs-extra');
+const upath = require('upath');
+const { getInstalledPath } = require('get-installed-path');
+const { exec } = require('child-process-promise');
+
+module.exports = {
+  generateLockFile,
+};
+
+async function generateLockFile(tmpDir) {
+  logger.debug(`Spawning pnpm install to create ${tmpDir}/shrinkwrap.yaml`);
+  let lockFile = null;
+  let stdout;
+  let stderr;
+  try {
+    const startTime = process.hrtime();
+    let cmd;
+    try {
+      // See if renovate is installed locally
+      const installedPath = upath.join(
+        await getInstalledPath('pnpm', {
+          local: true,
+        }),
+        'lib/bin/pnpm.js'
+      );
+      cmd = `node ${installedPath}`;
+    } catch (localerr) {
+      logger.debug('No locally installed pnpm found');
+      // Look inside globally installed renovate
+      try {
+        const renovateLocation = await getInstalledPath('renovate');
+        const installedPath = upath.join(
+          await getInstalledPath('pnpm', {
+            local: true,
+            cwd: renovateLocation,
+          }),
+          'lib/bin/pnpm.js'
+        );
+        cmd = `node ${installedPath}`;
+      } catch (nestederr) {
+        logger.debug('Could not find globally nested pnpm');
+        // look for global pnpm
+        try {
+          const installedPath = upath.join(
+            await getInstalledPath('pnpm'),
+            'lib/bin/pnpm.js'
+          );
+          cmd = `node ${installedPath}`;
+        } catch (globalerr) {
+          logger.warn('Could not find globally installed pnpm');
+          cmd = 'pnpm';
+        }
+      }
+    }
+    logger.debug(`Using pnpm: ${cmd}`);
+    cmd += ' install';
+    cmd += ' --shrinkwrap-only';
+    cmd += ' --ignore-scripts';
```

Comment:
I guess, also add `--ignore-pnpmfile`. `pnpmfile.js`, when available, is executed to configure the [hooks](https://pnpm.js.org/docs/usage/hooks.html). As it is a js file, maybe it can do problems on the server where renovate runs?

Reply:
Thanks. I wasn't sure if scripts or hooks would actually be run when generating shrinkwrap only. In theory they are not necessary just for that?

Reply:
scripts won't run but hooks will. Hooks are used to re-write manifests, so they would be used to create the `shrinkwrap.yaml`